### ISSUE TITLE
raw disk

### ISSUE THREAD

**Issue by boogerlad** [agentm/project-m36#337]
I see Project:M36 currently saves its state in memory or to a file. I don't know how hard it would be, but it would be nice to use an Intel 3D XPoint or some other sort of fast non volatile memory directly. Possibly better performance due to being able to optimize further, and operating system independent. This would also solve the chicken and the egg problem of new operating systems without hierarchical file systems.

**Comment by agentm** [comment]
Indeed, I am excited about new storage technologies as well. Project:M36 has a bunch of low-hanging fruit to implement in this context. I hope that you have seen "[UltraRAM](https://www.pcworld.com/article/564354/new-breakthrough-could-combine-storage-and-ram-into-a-single-component.html)" as well. It's not immediately clear, at least to me, how/when the operating system will discern between temporary and permanent storage and the current APIs will probably not suffice, so I'm keeping an eye on this.

**Comment by agentm** [comment]
Note that Project:M36 currently uses essential and common file system features:

* journaled fs metadata- enables durability of ACID
* fsync/fdatasync- enables durability of ACID
* parallel writing to disk
